### ISSUE TITLE
Add some way to clear cache without remounting

### ISSUE THREAD

**Issue by JuniorJPDJ** [openscopeproject/ZipROFS#6]
First thing that came up to my mind was to open some invisible virtual file in main directory.
eg. `echo "" > ./.clean_ziprofs_cache`
Of course `.clean_cache` wouldn't be listed in `readdir()` call

At the moment when I remount underlying filesystem (which was pretty often bcs of bugs in it) I need to remount also ZipROFS and restart all software which was using it.
Some workaroundish way to clean cache and reopen existing handles would be helpful.

**Comment by qu1ck** [comment]
Probably handling SIGHUP to dump cache is as close to a convention as we can get. At least it's better then creating some special meaning files.

By the way you will probably still have to restart software using ziprofs because it likely doesn't handle file handles becoming invalid well. Software that can handle that also survives ziprofs restart as I can attest from my everyday usage.

**Comment by JuniorJPDJ** [comment]
We could close any cached files and refresh handles inside ZipROFS to make this seamless

**Comment by JuniorJPDJ** [comment]
The solution I've in my mind is clearing cache and freezing ZipROFS process until underlying FS remounts

**Comment by qu1ck** [comment]
I don't think there is a way for ziprofs to distinguish between empty root and not mounted root. So freeze/unfreeze triggers have to be external, you could use SIGUSR1/2 for that.

**Comment by JuniorJPDJ** [comment]
I meant just freezing whole process with `SIGSTOP` :D

**Comment by qu1ck** [comment]
That works too but then you can't clear cache while the process is frozen.

**Comment by JuniorJPDJ** [comment]
I would guess it would process `SIGUSR` before anything other after unfreezing.
We could also provide [`SIGTSTP`](https://www.gnu.org/software/libc/manual/html_node/Job-Control-Signals.html) as it is being handled by process - which would first clear cache, then freeze process.

**Comment by qu1ck** [comment]
> I would guess it would process SIGUSR before anything other after unfreezing.

Unless this is documented behavior I wouldn't count on it.

> We could also provide SIGTSTP as it is being handled by process - which would first clear cache, then freeze process.

I would rather go for more generic solution. Sometimes dumping cache is useful without freezing process, these operations shouldn't be tied together.

**Comment by JuniorJPDJ** [comment]
Yes, but it could be function called twice, for `SIGUSR` and `SIGTSTP`.

**Comment by qu1ck** [comment]
So this sounds like a plan:
1. Handle SIGHUP to drop cache
2. Handle SIGTSTP to freeze and drop cache
3. Handle SIGCONT to unfreeze and attempt to refresh file handles

For refreshing file handles I think we will have to try to refresh non-zip handles as well. For that we will need a mapping of ziprofs handles to system handles instead of current odd-even scheme.

But I probably won't have time to implement this stuff in a while.

**Comment by JuniorJPDJ** [comment]
We could just drop and zipfiles instead of holding next handles.
Those are just open ZipFiles which we can close and reopen using cached implementation if needed.
